xls/examples/parameterized/BUILD

```diff
@@ -0,0 +1,105 @@
+load("//xls/build_rules:xls_utilities.bzl", "proto_to_dslx")
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+load(
+    "//xls/build_rules:xls_build_defs.bzl",
+    "xls_dslx_ir",
+    "xls_dslx_library",
+    "xls_ir_opt_ir",
+    "xls_ir_verilog",
+)
+
+proto_library(
+    name = "proto_schema_lib",
+    srcs = [":param.proto"],
+)
+
+# Parameters for the narrow variant of the design
+proto_to_dslx(
+    name = "parameters_narrow",
+    proto_schema = ":proto_schema_lib",
+    proto_name = "xls.DesignVariants",
+    textproto = ":narrow.textproto",
+)
+
+xls_dslx_library(
+    name = "parameters_narrow_lib",
+    srcs = [":parameters_narrow"],
+)
+
+# Parameters for the wide variant of the design
+proto_to_dslx(
+    name = "parameters_wide",
+    proto_schema = ":proto_schema_lib",
+    proto_name = "xls.DesignVariants",
+    textproto = ":wide.textproto",
+)
+
+xls_dslx_library(
+    name = "parameters_wide_lib",
+    srcs = [":parameters_wide"],
+)
+
+# Design implementation rules - narrow variant
+xls_dslx_library(
```

Comment:
I think we can have the parameters be included directly in the library.

Reply:
Please, see comment below.